### ISSUE TITLE
Notification subscription not working

### ISSUE THREAD

**Issue by collimarco** [mozilla-mobile/fenix#18144]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce

Visit https://pushpad.xyz/demo and try to send a notification.

### Expected behavior

A permission prompt is displayed and the notification is successfully sent. 

### Actual behavior

An error message "Notifications are blocked" is returned.

Note that the same code / demo works properly on Firefox (Windows, Mac), Chrome (Windows, Mac, Android), Edge, etc. So this is definitely a bug specific to Firefox on Android.

If you can't reproduce this bug please also try Settings -> Delete navigation data and clear everything (I noticed this bug after doing that, so it may be related).

### Device information

* Device vendor / model and Android version: Samsung Galaxy S10+, latest Android version (11)
* Firefox for Android version: latest (v85)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-9800)

**Comment by test491** [comment]
#### etp settings
custom - cryptominers, fingerprinters, redirect trackers, tracking content - all tabs, cookies - all cookies

#### reproducible with above etp settings

* version: `86.1.1`
* version: `87.0.0-beta.1 (Build #2015795065)`
* version: `Nightly 210228 (Build #2015795881)`

#### notes
after tapping send, it just shows message `Sending...`

#### android 9, Galaxy J7 Nxt

**Comment by test491** [comment]
#### not reproducible

`chrome 88.0.4324.93`

it shows message: Notifications are blocked. Please open your browser preferences or click the lock near the address bar to change your notification preferences.

screenshot of above message:
![screenshot_chrome](https://user-images.githubusercontent.com/78364929/109428139-abcd5b80-7a1b-11eb-9dff-d9e9ea246db8.jpg)

* settings
notifications -> on
notification style -> sound and pop up
screenshot of notification settings:
![screenshot_settings](https://user-images.githubusercontent.com/78364929/109428125-99ebb880-7a1b-11eb-895e-3606656265c2.jpg)

#### android 9, Galaxy J7 Nxt

**Comment by collimarco** [comment]
@test491 You are using **Private mode** on Chrome! Notifications are blocked by Chrome in private mode: that's why it doesn't work.

Please test without private mode enabled, because browsers have a different behavior.

If you don't use private mode, you will see that the demo works properly on Firefox (Windows, Mac), Chrome (Windows, Mac, Android), Edge, etc. 

However Firefox on Android doesn't work...

**Comment by test491** [comment]
hello @collimarco 

nice observation. however, i rarely use chrome. by defualt, for all apps i activate "dark theme" if available. pbm logo in screenshot belongs to firefox. see fresh screenshot
![today](https://user-images.githubusercontent.com/78364929/109453143-5030bb80-7a77-11eb-98ca-635936bcf029.jpg)

**Comment by collimarco** [comment]
@test491 

Thanks for looking into this bug.

In your screenshot there are two "private mode" icons: one is the Chrome icon (on the left) and one is the Firefox icon (on the right). So you probably have the private mode enabled in both browsers.

If Pushpad didn't work on Chrome on Android we would notice that and we would have many reports: Chrome on Android represents the majority of our subscriptions so I am 100% sure that it works... I also tested it right now and it works (Chrome 87 on Android 11). Chrome on Android has been supported for 5 years at least.

Please check the following settings on Chrome for Android:
- **you are not in private mode**
- **you have not blocked the notifications for all websites** (Chrome settings -> Website settings -> Notifications -> Websites can ask to send notifications)
- **you have not blocked the notifications for this domain** (click the lock near the address bar when there is the demo page open and check that notifications are not blocked there)
- **you have not blocked the notifications for the Chrome app itself** (from system settings)
- also try to **update Chrome** to the latest version to avoid strange bugs

**Comment by sflorean** [comment]
I was able to successfully receive notification in normal browsing state on latest Nightly on this devices: Samsung Note 10 (Android 11) and LG G7 Fit (Android 8.1).

I wasn't able to receive the notification when visiting the site in private browsing, same devices, I got a constant "sending mode".
[log.txt](https://github.com/mozilla-mobile/fenix/files/6236420/log.txt)

**Comment by collimarco** [comment]
This bug is still reproducible and must be kept open.

You can simply try to send a notification here: https://pushpad.xyz/demo

It works on every browser, including Firefox desktop, but not on Firefox on Android.

**Comment by collimarco** [comment]
The subscription seems to work now, but when you click on the notification nothing happens.

**Comment by sflorean** [comment]
Tested again and tapping on the notification opens the browser. 
@collimarco could you give another check? thank you!

**Comment by collimarco** [comment]
@sflorean It opens the browser, but not the website / web page that the notification should open...

**Comment by collimarco** [comment]
Keep open

**Comment by csadilek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1809283

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
